tools/test-npm.sh

```diff
@@ -37,7 +38,7 @@ node cli.js rebuild
 # install npm devDependencies and run npm's tests
 node cli.js install --ignore-scripts
 # run the tests
-node cli.js run-script test-node
+node cli.js run-script test-node -- --reporter=tap | tee ../test-npm.tap
```

Comment:
@TheAlphaNerd So you're saying have a `./configure` option of `-p tap` or `--reporter tap` and `--logfile test.tap`, and Makefile/vcbuild.bat variables called `REPORTER` and `LOGFILE`?

Follow up: `--reporter` or `--progress`? I'm leaning towards progress as it's what [tools/test.py](https://github.com/nodejs/node/blob/master/tools/test.py#L1309) does.